neqo-http3/src/features/extended_connect/tests/webtransport/mod.rs

```diff
@@ -78,28 +78,28 @@ fn connect_with(client: &mut Http3Client, server: &mut Http3Server) {
     let out = client.process(None, now());
     assert_eq!(client.state(), Http3State::Initializing);
 
-    let out = server.process(out.dgram(), now());
```

Comment:
It's possible that we could make `dgram()` return a reference as well now and remove the reference function.  Would that help?

Reply:
It would reduce the diff of the PR. That said, `as_dgram_ref` is the more [Rust idiomatic name](https://rust-lang.github.io/api-guidelines/naming.html). Thus I suggest keeping the methods (`dgram`, `as_dgram_ref`) as is.

Not a strong opinion. Your call @martinthomson.